hoomd/hpmc/pytest/test_shape_updater.py

```diff
@@ -234,15 +234,109 @@ def __call__(self, type_id, param_list):
     #  - shape and params should change
     #  - volume should remain unchanged
     move.param_move_probability = 1
-    sim.run(10)
-    assert np.sum(updater.shape_moves) == 20
+    sim.run(20)
+    assert np.sum(updater.shape_moves) == 40
+
+    # Check that the shape parameters have changed
     assert not np.allclose(mc.shape["A"]["a"], ellipsoid["a"])
     assert not np.allclose(mc.shape["A"]["b"], ellipsoid["b"])
     assert not np.allclose(mc.shape["A"]["c"], ellipsoid["c"])
     assert not np.allclose(move.params["A"], [1])
+
+    # Check that the shape parameters map back to the correct geometry
+    assert np.allclose(move.params["A"], [
+        mc.shape["A"]["a"] / mc.shape["A"]["b"],
+        mc.shape["A"]["a"] / mc.shape["A"]["c"]
+    ])
+
+    # Check that the callback is conserving volume properly
     assert np.allclose(updater.particle_volumes, 4 * np.pi / 3)
 
 
+def test_python_callback_shape_move_pyramid(simulation_factory,
```

Comment:
To make sure this actually tests the new line of code that was added, we need to make sure the parameter values aren't updated when the trial shape move creates overlaps between particles. I don't believe this test does that right now.

Reply:
I can confirm that this test seems to pass regardless of whether the fix is implemented - but as far as I can tell I would expect the test to fail when trial moves are rejected. Can you confirm this behavior?

Reply:
I am not able to reproduce that behavior with a distance of 3.0. I can reproduce it at a distance of 2.5 though.

Reply:
On revivification I have the same behavior - this is fixed in [70f0472](https://github.com/glotzerlab/hoomd-blue/pull/1696/commits/70f0472ad7840b4e28acc8e3859bbaf25b88640f)